definitions/checks/foreman_facts_names.rb

```diff
@@ -0,0 +1,29 @@
+class Checks::ForemanFactsNames < ForemanMaintain::Check
+  metadata do
+    label :foreman_fact_names
+    tags :default
+    confine do
+      feature(:foreman_database)
+    end
+    description 'Check number of fact names in database'
+  end
+
+  def run
+    max = 10_000
+    sql = <<-SQL
+    select fact_values.host_id, count(fact_values.id) from fact_values
+    group by fact_values.host_id order by count desc limit 1
+    SQL
+    result = feature(:foreman_database).query(sql).first
+    if result
+      host_id = result['host_id']
+      fact_values = result['fact_values']
+    else
+      fact_values = 0
```

Comment:
You can remove `else` part here & run assert inside `if`.

Reply:
Yes